src/test/java/in/projecteka/consentmanager/clients/PatientServiceClientTest.java

```diff
@@ -35,7 +36,7 @@ public void init() {
         WebClient.Builder webClientBuilder = WebClient.builder()
                 .exchangeFunction(exchangeFunction);
         LinkServiceProperties serviceProperties = new LinkServiceProperties("http://user-service/");
-        patientServiceClient = new PatientServiceClient(webClientBuilder, serviceProperties);
+        patientServiceClient = new PatientServiceClient(webClientBuilder, () -> Mono.just(string()), serviceProperties.getUrl());
```

Comment:
It could have been better if there is a check about, is that the same token is being passed to the downstream service?

Reply:
WebClientBuilder is mocked with an exchangeFunction. This string() is passed In the header as authorization token to the downstream service. 

I am not sure how to link this string() to that exchangeFunction in order to assert that the header is called with the token that was returned by tokenGenerator. Is there a way to do this?

Reply:
argument captor, capture the request and validate the header.